carpet/lista_analysis.py

```diff
@@ -124,57 +71,75 @@ def __init__(self, A, n_layers, initial_parameters=None, learn_th=True,
                          verbose=verbose, device=device)
 
     def get_global_parameters(self, initial_parameters):
-        initial_parameters_prox = unravel_group_params(
-            initial_parameters.get('prox', {})
-        )
 
         if self.learn_prox in (LEARN_PROX_GLOBAL, LEARN_PROX_FALSE):
+            initial_parameters_prox = unravel_group_params(
+                initial_parameters.get('prox', {})
+            )
             self.prox_tv = ListaLASSO(
                 A=self.I_k, n_layers=self.n_inner_layers,
                 learn_th=True, name="Prox-TV-Lista",
                 initial_parameters=initial_parameters_prox,
                 device=self.device)
 
-        if self.learn_prox == LEARN_PROX_GLOBAL:
-            self._register_parameters(
-                ravel_group_params(self.prox_tv.parameter_groups),
-                group_name='prox'
-            )
-            self.force_learn_groups.append('prox')
+            if self.learn_prox == LEARN_PROX_GLOBAL:
+                self._register_parameters(
+                    ravel_group_params(self.prox_tv.parameter_groups),
+                    group_name='prox'
+                )
+                self.force_learn_groups.append('prox')
 
-        if self.learn_prox == LEARN_PROX_PER_LAYER:
+        elif self.learn_prox == LEARN_PROX_PER_LAYER:
             # Make sure that all prox have been correctly initialized with
             # initial parameters
             for layer_id in range(self.n_layers):
                 if layer_id in self.prox_tv:
                     continue
-                group_name = f'layer-{layer_id}'
+                layer_params = self.parameter_groups[f'layer-{layer_id}']
                 initial_parameters_prox = unravel_group_params(
-                    {k.split(':', 1)[1]: v
-                     for k, v in self.parameter_groups[group_name].items()
+                    {k.split(':', 1)[1]: v for k, v in layer_params.items()
                      if k.startswith('prox:')}
                 )
-                self.prox_tv[layer_id] = ListaLASSO(
-                    A=self.I_k, n_layers=self.n_inner_layers, learn_th=True,
-                    name=f"Prox-TV-Lista[layer={layer_id}]",
-                    initial_parameters=initial_parameters_prox,
-                    device=self.device
+                self._initialize_prox_tv_per_layer(
+                    layer_id, layer_params, initial_parameters_prox
                 )
+        else:
```

Comment:
Yes, it spotted it was missing in previous commit. Good

Reply:
On PR, I usually only look at the global diff.
I don't try to keep informative commits as it takes too much time and does not bring a lot of benefit.